src/SideNavigation/SideNavigation.js

```diff
@@ -1,204 +1,251 @@
-                                                            {ch.name}
-                                                        </Link>
-                                                    ) : null}
-
-                                                    {ch.url ? (
-                                                        <a
-                                                            className={this.getSubLinkClasses(ch.id)}
-                                                            href={ch.url}
-                                                            key={ch.id}
-                                                            onClick={e => this.handleSelectChild(e, ch.id)}>
-                                                            {ch.name}
-                                                        </a>
-                                                    ) : null}
-                                                </React.Fragment>
-                                            );
-                                        })}
-                                    </ul>
-                                ) : null}
-                            </li>
-                        );
-                    })}
-                </ul>
-            </BrowserRouter>
+    const sideNavList = (
+        <ul
+            {...rest}
+            aria-expanded={hasParent && open}
+            aria-hidden={hasParent && !open}
+            className={sideNavListClasses}>
+            {React.Children.map(children, (child) => {
+                if (React.isValidElement(child)) {
+                    return React.cloneElement(child, {
+                        isSubItem: hasParent,
+                        onItemSelect: onItemSelect,
+                        selected: selectedId === child.props.id,
+                        selectedId: selectedId
+                    });
+                } else {
+                    return child;
+                }
+            })}
+        </ul>
+    );
+
+    if (title && !hasParent) {
+        return (
+            <div
+                className={sideNavHeaderlasses}>
+                <h1 {...titleProps} className='fd-side-nav__title'>
+                    {title}
+                </h1>
+                {sideNavList}
+            </div>
         );
     }
-}
+
+    return sideNavList;
+};
+
 SideNavList.propTypes = {
-    items: PropTypes.array.isRequired,
-    className: PropTypes.string
+    children: PropTypes.element,
+    className: PropTypes.string,
+    hasParent: PropTypes.bool,
+    open: PropTypes.bool,
+    selectedId: PropTypes.string,
+    title: PropTypes.string,
+    titleProps: PropTypes.object,
+    onItemSelect: PropTypes.func
 };
 
 SideNavList.propDescriptions = {
-    items: 'An array of objects with keys \'id\', \'url\', \'name\', \'hasChild\', \'child\', and \'glyph\' setting the attributes of the items.'
+    hasParent: 'Automatically set to **true** when item has a parent `SideNavListItem`',
```

Comment:
if this is set automatically, should it be exposed as a prop type?

Reply:
We have an issue where internal props have to be documented or they are left with a blank description. @greg-a-smith and I chatted about this and decided to write that these props are used for internal use only. I've opened #393 which will aim to correct this issue and would remove these prop descriptions.